pkg/recipes/driver/types.go

```diff
@@ -42,6 +42,13 @@ type Driver interface {
 	GetRecipeMetadata(ctx context.Context, opts BaseOptions) (map[string]any, error)
 }
 
+type DriverWithSecrets interface {
+	Driver
+
+	// FindSecretIDs gets the secret ID references from environment definition.
```

Comment:
Could you add more details in this comment? What is the secret ID for example? Where does it read it from in the environment definition - recipe config. Why is it needed - to access private modules (for now, with providers change it would extend to secrets passed into the providers config as well)